jwk.go

```diff
@@ -73,6 +76,8 @@ func (k JsonWebKey) MarshalJSON() ([]byte, error) {
 		raw, err = fromEcPrivateKey(key)
 	case *rsa.PrivateKey:
 		raw, err = fromRsaPrivateKey(key)
+	case *x509.Certificate:
+		raw, err = fromX509Certificate(key)
```

Comment:
I don't think that's what we want -- I don't see the reasoning behind allowing the `Key` field in `JsonWebKey` to support values of type `*x509.Certificate`. This makes it impossible to properly attach a chain of certificates to a key, you would be limited to one. The certificate chain doesn't replace the key, it just supplements it. What you want is to have a `Certificates` field (of type `[]*x509.Certificate`) in `JsonWebKey` which then gets marshaled into `X5c` in the JSON representation of the JWK.